python/src/scipp/compat/mantid.py

```diff
@@ -193,17 +193,119 @@ def validate_and_get_unit(unit):
         return known_units[unit]
 
 
-def init_pos(ws):
-    nHist = ws.getNumberHistograms()
-    pos = np.zeros([nHist, 3])
+def _to_array(positions, idx, pos, spec_idx):
+    positions[idx, :] = np.array([spec_idx, pos.X(), pos.Y(), pos.Z()])
+
+
+def _to_spherical(input):
+    transformed = np.ones(shape=input.shape)
+    transformed[:, 0] = input[:, 0]  # copy metadata
+    transformed[:, 1] = np.sqrt(np.sum(input[:, 1:]**2, axis=1))  # r
+    transformed[:, 2] = np.arccos(input[:, 3] /
+                                  transformed[:, 1])  # arccos ( z / r)
+    transformed[:, 3] = np.arctan2(input[:, 2], input[:, 1])  # arctan2(y, x)
```

Comment:
Weird use of numpy arrays (why transposed?). Just use multiple outputs, or a dataset?

Reply:
Nothing is transposed here.

It's probably more convenient to do the transform as part of a scipp dataset.

Reply:
There are a few missing trig functions I need to add to `variable.cpp` to get this to work fully in scipp.